### ISSUE TITLE
Bug with consecutive branch.

### ISSUE THREAD

**Issue by Hoblovski** [riscv-and-rust-and-decaf/riscv32i-cpu-chisel#2]
For the assembly below
~~~
    li x1, 0xC0FFEE00
    addi x2, x0, 0x100
    addi x3, x0, 0x400

l1:
    sw x1, 0(x2)
    addi x2, x2, 4
    blt x2, x3, l1

l2:
    j l2
    nop
    nop
~~~

Implementation on master (e445360) fails to behave correctly.
It works as if both branch hadn't been taken.

**Comment by Hoblovski** [comment]
The problem occurs because IF is stalled when ID gives the branch signal

**Comment by Hoblovski** [comment]
The final cause is with read-after-load stalling.
Even if the read value is not used, stalling occurred.
Fixed in 197888c
